pkg/builder/devcontainer.go

```diff
@@ -0,0 +1,384 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package builder
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/daytonaio/daytona/pkg/builder/devcontainer"
+	"github.com/daytonaio/daytona/pkg/docker"
+	"github.com/daytonaio/daytona/pkg/logger"
+	"github.com/daytonaio/daytona/pkg/workspace"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+)
+
+type BuildOutcome struct {
+	Outcome               string `json:"outcome"`
+	ContainerId           string `json:"containerId"`
+	RemoteUser            string `json:"remoteUser"`
+	RemoteWorkspaceFolder string `json:"remoteWorkspaceFolder"`
+}
+
+type DevcontainerBuilderConfig struct {
+	buildId                      string
+	project                      workspace.Project
+	loggerFactory                logger.LoggerFactory
+	localContainerRegistryServer string
```

Comment:
Maybe we should remove the `local` part of this to open up possibility of defining your own registry?

Reply:
I was thinking about it as well, but would rather leave it explicit and implement custom registries in the separate PR.